src/components/Card/Card.test.tsx

```diff
@@ -1,37 +1,98 @@
+    const { container } = render(<Card {...DEFAULT_PROPS} title={'1,'} value={'2,'} bottomText={'3'} />);
+    expect(container).toHaveTextContent('1,2,3');
+  });
 
-test('should work with synchronous data', () => {
-  const { getByText } = renderComponent(DEFAULT_PROPS);
-  expect(getByText('Title Text')).toBeInTheDocument();
-  expect(getByText('123')).toBeInTheDocument();
-  expect(getByText('Bottom Text')).toBeInTheDocument();
-});
+  describe('semantic tags', () => {
+    test('should render title as <label>', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} title={'mock title'} />);
+      const label = container.getElementsByTagName('label');
+      expect(label.length).toBe(1);
+      expect(label[0]).toHaveTextContent('mock title');
+    });
+  });
 
-test('should show skeleton when loading asynchronous data', () => {
-  const { getByText } = renderComponent({ ...DEFAULT_PROPS, value: { isLoading: true } });
-  expect(getByText('Skeleton')).toBeInTheDocument();
-});
+  describe('class names', () => {
+    test('should apply Title class to only the title element', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} title={'mock title'} />);
+      const title = container.getElementsByClassName(styles.Title);
+      expect(title.length).toBe(1);
+      expect(title[0]).toHaveTextContent('mock title');
+    });
+
+    test('should only apply Value class to only the value element', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} value={'mock value'} />);
+      const value = container.getElementsByClassName(styles.Value);
+      expect(value.length).toBe(1);
+      expect(value[0]).toHaveTextContent('mock value');
+    });
+
+    test('should only apply BottomText class to the bottom text element', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} bottomText={'mock bottom text'} />);
+      const bottomText = container.getElementsByClassName(styles.BottomText);
+      expect(bottomText.length).toBe(1);
+      expect(bottomText[0]).toHaveTextContent('mock bottom text');
+    });
+  });
+
+  describe('when text props are synchronous', () => {
+    test('should render title', () => {
+      render(<Card {...DEFAULT_PROPS} title={'mock title'} />);
+      expect(screen.getByText('mock title')).toBeInTheDocument();
+    });
+
+    test('should render value in the correct element (<b>)', () => {
+      render(<Card {...DEFAULT_PROPS} value={'mock value'} />);
+      expect(screen.getByText('mock value')).toBeInTheDocument();
+    });
+
+    test('should render synchronous bottomText', () => {
+      render(<Card {...DEFAULT_PROPS} bottomText={'mock bottom text'} />);
+      expect(screen.getByText('mock bottom text')).toBeInTheDocument();
+    });
+  });
+
+  describe('when bottom text is asynchronous', () => {
+    test('should show loading skeleton when loading bottom text', () => {
```

Comment:
do you think in scenarios like this we should check that bottom text is falsy when loading? Not sure how far we want to take coverage

similar to below, would you expect to assert that skeleton is falsy when text/value is truthy

Reply:
Yeah probably worth checking the Skeleton is no longer in the DOM - I'll add it to the tests.